Submitted By: Ken Moffat <ken at linuxfromscratch dot org>
Date: 2019-08-17
Initial Package Version: 3.4.6
Upstream Status: Uncertain
Origin: Found at fedora, originally for 3.4.4 with new glibc.
Description: With glibc-2.30 some RES macros have been dropped, package FTBFS.

dns_str_resflags.c:63:22: error: 'RES_INSECURE1' undeclared here (not in a function); did you mean 'RES_RECURSE'?

diff --git a/src/dns/dns_str_resflags.c b/src/dns/dns_str_resflags.c
index 5f2cce5..df32345 100644
--- a/src/dns/dns_str_resflags.c
+++ b/src/dns/dns_str_resflags.c
@@ -60,10 +60,16 @@ static const LONG_NAME_MASK resflag_table[] = {
     "RES_DEFNAMES", RES_DEFNAMES,
     "RES_STAYOPEN", RES_STAYOPEN,
     "RES_DNSRCH", RES_DNSRCH,
+#ifdef RES_INSECURE1
     "RES_INSECURE1", RES_INSECURE1,
+#endif
+#ifdef RES_INSECURE2
     "RES_INSECURE2", RES_INSECURE2,
+#endif
     "RES_NOALIASES", RES_NOALIASES,
+#ifdef RES_USE_INET6
     "RES_USE_INET6", RES_USE_INET6,
+#endif
 #ifdef RES_ROTATE
     "RES_ROTATE", RES_ROTATE,
 #endif
